x/config/shadowsocks.go

```diff
@@ -144,11 +144,14 @@ func parseShadowsocksSIP002URL(url *url.URL) (*shadowsocksConfig, error) {
 		return nil, errors.New("host not specified")
 	}
 	config.serverAddress = url.Host
-	cipherInfoBytes, err := base64.URLEncoding.WithPadding(base64.NoPadding).DecodeString(url.User.String())
-	if err != nil {
-		return nil, fmt.Errorf("failed to decode cipher info [%v]: %w", url.User.String(), err)
-	}
-	cipherName, secret, found := strings.Cut(string(cipherInfoBytes), ":")
+	cipherInfo := url.User.String()
+	// Cipher info can be optionally encoded with Base64URL.
+	encoding := base64.URLEncoding.WithPadding(base64.NoPadding)
+	decodedCipherInfo, err := encoding.DecodeString(cipherInfo)
+	if err == nil && encoding.EncodeToString(decodedCipherInfo) == cipherInfo {
```

Comment:
Why do you need to re-encode the string? The decoding already succeeded.

Reply:
Whoops I was trying something else earlier and forgot to remove. Thanks.

Reply:
Actually I remember now. I wanted to protect against strings that look like valid base64 (i.e. they decode fine) but aren't. Re-encoding would catch some of those, since the encoded version would differ from the original. However, on second thought I don't think that's really necessary, since in practice we'd expect unecoded strings to contain a colon. I'll leave it off.